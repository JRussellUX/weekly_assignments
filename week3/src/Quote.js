function Quote(props) {
    const quote = props.quote;

    return (
        quote ? (<h3>
          <span>â€œ</span>
          {quote.quote} ({quote.character})
          <span>"</span>
        </h3>) : null
    
  );
}

export default Quote;
