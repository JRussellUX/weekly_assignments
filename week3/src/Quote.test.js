import Quote from './Quote.js';
import renderer from 'react-test-renderer';
import {render, screen} from '@testing-library/react';

test('renders quote screen', () => {
  let testQuote = {quote: "Test Quote", character: "Test Character"};
  const quote = renderer.create(<Quote quote={testQuote}/>);
  let tree = quote.toJSON();
  expect(tree).toMatchSnapshot();
});

it('Quote appears', () => {
  let testQuote = {quote: "Test Quote", character: "Test Character"};
  render(<Quote quote={testQuote}/>);
  const actualQuote = screen.getByText("Test Quote (Test Character)");
  expect(actualQuote).toBeTruthy();
});