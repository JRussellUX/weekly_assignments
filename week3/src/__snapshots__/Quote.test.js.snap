// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders quote screen 1`] = `
<h3>
  <span>
    â€œ
  </span>
  Test Quote
   (
  Test Character
  )
  <span>
    "
  </span>
</h3>
`;
